realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -0,0 +1,371 @@
+#include <object-store/src/object_store.hpp>
+#include "io_realm_internal_SharedRealm.h"
+
+#include "shared_realm.hpp"
+#include "util.hpp"
+
+using namespace realm;
+
+static_assert(SchemaMode::Automatic ==
+                      static_cast<SchemaMode>(io_realm_internal_SharedRealm_SCHEMA_MODE_VALUE_AUTOMATIC), "");
+static_assert(SchemaMode::ReadOnly ==
+                      static_cast<SchemaMode>(io_realm_internal_SharedRealm_SCHEMA_MODE_VALUE_READONLY), "");
+static_assert(SchemaMode::ResetFile ==
+                      static_cast<SchemaMode>(io_realm_internal_SharedRealm_SCHEMA_MODE_VALUE_RESET_FILE), "");
+static_assert(SchemaMode::Additive ==
+                      static_cast<SchemaMode>(io_realm_internal_SharedRealm_SCHEMA_MODE_VALUE_ADDITIVE), "");
+static_assert(SchemaMode::Manual ==
+              static_cast<SchemaMode>(io_realm_internal_SharedRealm_SCHEMA_MODE_VALUE_MANUAL), "");
+
+JNIEXPORT jlong JNICALL
+Java_io_realm_internal_SharedRealm_nativeCreateConfig(JNIEnv *env, jclass, jstring realm_path, jbyteArray key,
+        jbyte schema_mode, jboolean in_memory, jboolean cache, jboolean disable_format_upgrade,
+        jboolean auto_change_notification)
+{
+    TR_ENTER()
+
+    try {
+        JStringAccessor path(env, realm_path); // throws
+        JniByteArray key_array(env, key);
+        Realm::Config *config = new Realm::Config();
+        config->path = path;
+        config->encryption_key = key_array;
+        config->schema_mode = static_cast<SchemaMode>(schema_mode);
+        config->in_memory = in_memory;
+        config->cache = cache;
+        config->disable_format_upgrade = disable_format_upgrade;
+        config->automatic_change_notifications = auto_change_notification;
+        return reinterpret_cast<jlong>(config);
+    } CATCH_STD()
+
+    return static_cast<jlong>(NULL);
+}
+
+JNIEXPORT void JNICALL
+Java_io_realm_internal_SharedRealm_nativeCloseConfig(JNIEnv *, jclass, jlong config_ptr)
+{
+    TR_ENTER_PTR(config_ptr)
+
+    auto config = reinterpret_cast<realm::Realm::Config*>(config_ptr);
```

Comment:
I don't have a strong opinion. So fine with me. Having it explicit also means that using auto still keeps it readable. eg`auto config = C(config_ptr)`is probably a bit too much magic.